rotkehlchen/tests/unit/globaldb/test_globaldb.py

```diff
@@ -1012,36 +1015,128 @@ def test_general_cache(globaldb):
 
     # check that deletion works properly
     with globaldb.conn.write_ctx() as write_cursor:
-        globaldb_delete_general_cache(
+        globaldb_delete_cache(
             write_cursor=write_cursor,
-            key_parts=[GeneralCacheType.CURVE_POOL_ADDRESS, '123'],
+            key_parts=[GeneralCacheType.CURVE_LP_TOKENS, '123'],
         )
-        values_5 = globaldb_get_general_cache_values(
+        values_5 = globaldb_get_cache_values(
             cursor=write_cursor,
-            key_parts=[GeneralCacheType.CURVE_POOL_ADDRESS, '123'],
+            key_parts=[GeneralCacheType.CURVE_LP_TOKENS, '123'],
         )
         assert len(values_5) == 0
-        values_6 = globaldb_get_general_cache_values(
+        values_6 = globaldb_get_cache_values(
             cursor=write_cursor,
-            key_parts=[GeneralCacheType.CURVE_POOL_ADDRESS, '456'],
+            key_parts=[GeneralCacheType.CURVE_LP_TOKENS, '456'],
         )
         assert len(values_6) == 2   # should have not been touched by the deletion above
-        globaldb_delete_general_cache(
+        globaldb_delete_cache(
             write_cursor=write_cursor,
-            key_parts=[GeneralCacheType.CURVE_POOL_ADDRESS, '456'],
+            key_parts=[GeneralCacheType.CURVE_LP_TOKENS, '456'],
         )
-        values_7 = globaldb_get_general_cache_values(
+        values_7 = globaldb_get_cache_values(
             cursor=write_cursor,
-            key_parts=[GeneralCacheType.CURVE_POOL_ADDRESS, '456'],
+            key_parts=[GeneralCacheType.CURVE_LP_TOKENS, '456'],
         )
         assert len(values_7) == 0
-        values_8 = globaldb_get_general_cache_values(
+        values_8 = globaldb_get_cache_values(
             cursor=write_cursor,
             key_parts=[GeneralCacheType.CURVE_POOL_TOKENS, '123'],
         )
         assert values_8 == values_1
 
 
+@pytest.mark.parametrize('globaldb_upgrades', [[]])
+@pytest.mark.parametrize('run_globaldb_migrations', [False])
+def test_unique_cache(globaldb):
+    """
+    Test that the added unique cache table in the globaldb works properly.
+    Tests insertion, deletion and reading.
+    """
+    ts_test_start = ts_now()
+    with globaldb.conn.write_ctx() as write_cursor:
+        # write some values
+        globaldb_set_cache_values(
+            write_cursor=write_cursor,
+            key_parts=[GeneralCacheType.CURVE_POOL_ADDRESS],
```

Comment:
The tests should be as close to reality as possible. Curve pool address should never stand alone as as a key. It's always pairned with the address.